src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -411,6 +421,7 @@ def csv(self, path: str) -> DataFrame:
         Returns:
             a :class:`DataFrame` that is set up to load data from the specified CSV file(s) in a Snowflake stage.
         """
+        path = _validate_stage_path(path)
```

Comment:
I think we need a changelog for this behavior that the path is eagerly validated